src/modules/cpu/kernel/ricap.hpp

```diff
@@ -0,0 +1,1629 @@
+
+        for (int i = 0; i < 4; i++)
+        {
+            roiPtrInput[i] = &roiPtrInputCropRegion[i];
+            compute_roi_validation_host(roiPtrInput[i], &roi[i], &roiDefault, roiType);
+            srcPtrImage[i] = srcPtr + (permutedIndices[permutedCount + i] * srcDescPtr->strides.nStride);
+            bufferLength[i] = roi[i].xywhROI.roiWidth * layoutParams.bufferMultiplier;
+            srcPtrChannel[i] = srcPtrImage[i] + (roi[i].xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi[i].xywhROI.xy.x * layoutParams.bufferMultiplier);
+            alignedLength[i] = (bufferLength[i] / 48) * 48;
+        }
+
+        Rpp8u *dstPtrImage, *dstPtrChannel;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+        dstPtrChannel = dstPtrImage;
+        Rpp32u vectorIncrement = 48;
+        Rpp32u vectorIncrementPerChannel = 16;
+
+        // Ricap with fused output-layout toggle (NHWC -> NCHW)
+        if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NCHW))
+        {
+            Rpp8u *srcPtrRow1, *srcPtrRow2, *srcPtrRow3, *srcPtrRow4, *dstPtrRowR, *dstPtrRowG, *dstPtrRowB;
+            srcPtrRow1 = srcPtrChannel[0];
+            srcPtrRow2 = srcPtrChannel[1];
+            srcPtrRow3 = srcPtrChannel[2];
+            srcPtrRow4 = srcPtrChannel[3];
+            dstPtrRowR = dstPtrChannel;
+            dstPtrRowG = dstPtrRowR + dstDescPtr->strides.cStride;
+            dstPtrRowB = dstPtrRowG + dstDescPtr->strides.cStride;
+
+            for (int i = 0; i < roi[0].xywhROI.roiHeight; i++)
+            {
+                Rpp8u *srcPtrTemp1, *srcPtrTemp2, *dstPtrTempR, *dstPtrTempG, *dstPtrTempB;
+                srcPtrTemp1 = srcPtrRow1;
+                srcPtrTemp2 = srcPtrRow2;
+                dstPtrTempR = dstPtrRowR;
+                dstPtrTempG = dstPtrRowG;
+                dstPtrTempB = dstPtrRowB;
+
+                int vectorLoopCount1 = 0;
+                for (; vectorLoopCount1 < alignedLength[0]; vectorLoopCount1 += vectorIncrement)
+                {
+                    __m128i p[3];
+                    rpp_simd_load(rpp_load48_u8pkd3_to_u8pln3, srcPtrTemp1, p);                             // simd loads
+                    rpp_simd_store(rpp_store48_u8pln3_to_u8pln3, dstPtrTempR, dstPtrTempG, dstPtrTempB, p); // simd stores
+                    srcPtrTemp1 += vectorIncrement;
+                    dstPtrTempR += vectorIncrementPerChannel;
+                    dstPtrTempG += vectorIncrementPerChannel;
+                    dstPtrTempB += vectorIncrementPerChannel;
+                }
+                for (; vectorLoopCount1 < bufferLength[0]; vectorLoopCount1 += 3)
+                {
+                    *dstPtrTempR = srcPtrTemp1[0];
+                    *dstPtrTempG = srcPtrTemp1[1];
+                    *dstPtrTempB = srcPtrTemp1[2];
+                    srcPtrTemp1 += 3;
+                    dstPtrTempR++;
+                    dstPtrTempG++;
```

Comment:
@snehaa8 Please make common change in this file to use post-increment efficiently while assigning itself

Reply:
Sure